### PR TITLE
HTCONDOR-1407

### PR DESCRIPTION
**Pull request by Todd-L-Miller**
Leaving this one in draft state until HTCONDOR-1283 is merged, at which point we'll need to change the target branch accordingly.

# HTCondor Pull Request Checklist for internal reviewers

- [x] Verify that (GitHub thinks) the merge is clean. If it isn't, and you're confident you can resolve the conflicts, do so. Otherwise, send it back to the original developer.
- [x] Verify that the related Jira ticket exists and has a target version number and that it is correct.
- [x] Verify that the Jira ticket is in review status and is assigned to the reviewer.
- [x] Verify that the Jira ticket (HTCONDOR-xxx) is mentioned at the beginning of the title. Edit it, if not
- [x] Verify that the branch destination of the PR matches the target version of the ticket
- [x] Check for correctness of change
- [x] Check for regression test(s) of new features and bugfixes (if the feature doesn't require root)
- [x] Check for documentation, if needed
- [x] Check for version history, if needed
- [ ] Check BaTLab dashboard for successful build (https://batlab.chtc.wisc.edu/results/workspace.php) and test for either the PR or a workspace build by the developer that has the Jira ticket as a comment.
- [x] Check that each commit message references the Jira ticket (HTCONDOR-xxx)

## After the above
- Hit the merge button if the pull request is approved and it is not a security patch (security changes require 2 additional reviews)
- If the pull request is approved, take the ticket out of review state
- Assign JIRA Ticket back to the developer

### CODE REVIEW

File: src/condor_tools/htcondor_cli/annex_create.py [hunk 1]
```diff
@@ -632,13 +632,19 @@ def annex_inner_func(
...
+                f"No jobs for '{annex_name}' are in the queue. Use 'htcondor job submit --annex-name' to add them first."
+            )
+        logger.debug(
+            f"""Found {len(annex_jobs)} annex jobs matching 'TargetAnnexName == "{annex_name}"."""
         )
-    logger.debug(
-        f"""Found {len(annex_jobs)} annex jobs matching 'TargetAnnexName == "{annex_name}"."""
-    )
+
+    if test is not None and test == 2:
```
Comment (matyasselmeci) [inline src/condor_tools/htcondor_cli/annex_create.py:646]:
```suggestion
    if test == 2:
```

File: src/condor_tools/htcondor_cli/annex_create.py [hunk 2]
```diff
@@ -956,14 +962,14 @@ def annex_name_exists(annex_name):
 
 
 def annex_create(logger, annex_name, **others):
-    if others.get("test") is not True:
+    if others.get("test") is None:
```
Comment (matyasselmeci) [inline src/condor_tools/htcondor_cli/annex_create.py:965]:
```suggestion
    if not others.get("test"):
```
unless you care about distinguishing between `""` and `None`

File: src/condor_tools/htcondor_cli/annex_create.py [hunk 3]
```diff
@@ -956,14 +962,14 @@ def annex_name_exists(annex_name):
...
         if annex_name_exists(annex_name):
             raise ValueError(f"You've already created an annex named '{annex_name}'.  To request more resources, use 'htcondor annex add'.")
     return annex_inner_func(logger, annex_name, **others)
 
 
 def annex_add(logger, annex_name, **others):
-    if others.get("test") is not True:
+    if others.get("test") is None:
```
Comment (matyasselmeci) [inline src/condor_tools/htcondor_cli/annex_create.py:972]:
```suggestion
    if not others.get("test"):
```

File: src/condor_tools/htcondor_cli/annex_create.py (hunk 2 above)
Comment (matyasselmeci) [reply src/condor_tools/htcondor_cli/annex_create.py:965]: or in this case distinguishing between `0` and `None`?

File: src/condor_tools/htcondor_cli/annex.py [hunk 4]
```diff
@@ -125,8 +125,8 @@ class Create(Verb):
         "test": {
             "args": ("--test",),
             "help": argparse.SUPPRESS,
-            "default": False,
-            "action": "store_true",
+            "type": int,
+            "default": None,
```
Comment (matyasselmeci) [inline src/condor_tools/htcondor_cli/annex.py:129]: --test now requires an int to follow it; have you made sure this doesn't impact any of your existing tests?

File: src/condor_tools/htcondor_cli/annex_create.py (hunk 2 above)
Comment (Todd-L-Miller) [reply src/condor_tools/htcondor_cli/annex_create.py:965]: Correct; I do want to distinguish between those; the test flag must not be present at all.

File: src/condor_tools/htcondor_cli/annex_create.py (hunk 3 above)
Comment (Todd-L-Miller) [reply src/condor_tools/htcondor_cli/annex_create.py:972]: See previous response.

**Review by matyasselmeci — APPROVED** [review]

**Pull request merged**